src/node_api.h

```diff
@@ -555,54 +157,16 @@ NAPI_EXTERN napi_status napi_queue_async_work(napi_env env,
 NAPI_EXTERN napi_status napi_cancel_async_work(napi_env env,
                                                napi_async_work work);
 
-// Methods for custom handling of async operations
-NAPI_EXTERN napi_status napi_async_init(napi_env env,
-                                        napi_value async_resource,
-                                        napi_value async_resource_name,
-                                        napi_async_context* result);
-
-NAPI_EXTERN napi_status napi_async_destroy(napi_env env,
-                                           napi_async_context async_context);
-
-NAPI_EXTERN napi_status napi_make_callback(napi_env env,
-                                           napi_async_context async_context,
-                                           napi_value recv,
-                                           napi_value func,
-                                           size_t argc,
-                                           const napi_value* argv,
-                                           napi_value* result);
-
 // version management
-NAPI_EXTERN napi_status napi_get_version(napi_env env, uint32_t* result);
-
 NAPI_EXTERN
 napi_status napi_get_node_version(napi_env env,
                                   const napi_node_version** version);
 
-// Promises
-NAPI_EXTERN napi_status napi_create_promise(napi_env env,
-                                            napi_deferred* deferred,
-                                            napi_value* promise);
-NAPI_EXTERN napi_status napi_resolve_deferred(napi_env env,
-                                              napi_deferred deferred,
-                                              napi_value resolution);
-NAPI_EXTERN napi_status napi_reject_deferred(napi_env env,
-                                             napi_deferred deferred,
-                                             napi_value rejection);
-NAPI_EXTERN napi_status napi_is_promise(napi_env env,
-                                        napi_value promise,
-                                        bool* is_promise);
-
 // Memory management
 NAPI_EXTERN napi_status napi_adjust_external_memory(napi_env env,
```

Comment:
I think this one could probably go into js_native_api.  It is not specific to Node and even platfroms without garbage collections might find the info useful (for reporting if nothing else) .  It is also an easy no-op if the platform does not want to do anything with it.